### ISSUE TITLE
A false negative about the rule SelfComparison

### ISSUE THREAD

**Issue by Zustin** [google/error-prone#2485]
Hi, when used Error Prone, I found a false negative about the rule [SelfComparison](https://errorprone.info/bugpattern/SelfComparison). The minimal case is below. Error prone should report a warning in method *func* because obj tries to invoke *compareTo* by itself.

```java
static class Pair {
   int a, b;
   public Pair(int a, int b) {
      this.a = a;
      this.b = b;
   }
   public int compareTo(Pair o) {
      if(o.a > this. a) {
         return -1;
      } else {
         if(o.a == this.a) {
            return 0;
         } else {
            return 1;
         }
      }
   }
}

public void func() {
  Pair p = new Pair(1, 2);
  p.compareTo(p);  // should report a warning here
}
```

Expected output: A warning
Actual output: No warnings

**Comment by sumitbhagwani** [comment]
This is because the class Pair isn't implementing Comparable interface. 
Without implementing the interface and implementing an @Override on compareTo, SelfComparison check cannot flag this

**Comment by sumitbhagwani** [comment]
https://github.com/google/error-prone/commit/e21a599944edfd1ff8929075eeba2e3540b685d0 should help in identifying such cases
